src/platform/packages/shared/kbn-opentelemetry-utils/src/flatten_to_attributes.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Attributes } from '@opentelemetry/api';
+import { isArray, isPlainObject } from 'lodash';
+
+/**
+ * Flattens a structured object in a way that is consistent with
+ * how OpenTelemetry defines {@link Attributes}
+ */
+export function flattenToAttributes(obj: Record<string, any>, parentKey: string = ''): Attributes {
```

Comment:
FMI: what's the difference with https://github.com/elastic/kibana/blob/c3fdb39dfb9784302c9ca398c81797b4c43a5dd3/src/platform/packages/shared/kbn-object-utils/src/flatten_object.ts#L22 ?

Same question for unflatten.

Reply:
that one assigns arrays, instead of recursively merging them.